### PR TITLE
Adding web hooks

### PR DESCRIPTION
**Pull request by nkatzman**
This PR adds in a github webhook endpoint

The endpoint only works for pull requests and syncs the pipeline every time

Adds todo's about what needs to be done next.  Part of #23

### CODE REVIEW

File: plugins/github.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+/**
+ * GitHub Webhook Plugin
+ *  - Validates that the event came from GitHub
+ *  - Opening a PR should sync the pipeline (creating the job) and start the new PR job
+ *  - Syncing a PR should stop the existing PR job and start a new one
+ *  - Closing a PR should stop the PR job and sync the pipeline (disabling the job)
+ * @method register
+ * @param  {Hapi.Server}    server
+ * @param  {Object}         options
```
Comment (tkyi) [inline plugins/github.js]: add more comments for `options.datastore`, `options.secret`

Comment (stjohnjohnson) [inline plugins/github.js]: Done

**Comment by tkyi** [comment]
👍  looks like a good first step

**Comment by d2lam** [comment]
👍

**Pull request merged**